From 0ba1beed963e6a374f207655233f46a4e724531e Mon Sep 17 00:00:00 2001
From: "Sukerkar, Amol N" <amol.n.sukerkar@intel.com>
Date: Tue, 21 Jun 2016 11:54:35 -0700
Subject: [PATCH 432/441] net: LitePath API Config Changes

Removed the extra CONFIG OPTION and moved the required CONFIGS
from arch/x86/Kconfig to drivers/net/Kconfig where they belong. Also
Added description for the new config values.

Signed-off-by: Sukerkar, Amol N <amol.n.sukerkar@intel.com>
---
 arch/x86/Kconfig              | 20 --------------------
 drivers/net/Kconfig           | 27 +++++++++++++++++++--------
 drivers/net/lpal_api/Makefile | 32 ++------------------------------
 3 files changed, 21 insertions(+), 58 deletions(-)

--- a/arch/x86/Kconfig
+++ b/arch/x86/Kconfig
@@ -467,26 +467,6 @@ menu "Intel Puma7 LitePath Support"
           This feature turns on the LitePath support logic for
           WiFi Access Point driver that wishes to use Partial
           Offload feature.
-
-    config LTQ_PPA_API_DIRECTPATH
-        bool "Enables Litepath API in Wave500"
-        depends on PPA_PUMA7
-        default n
-        ---help---
-           Configures Litepath feature for Wave500. This symbol
-           enables the WiFi Access Point driver to make use of
-           the Directpath API exposed by LitePath for Partial
-           Offload.
-
-    config LTQ_PPA_API_DIRECTPATH_HAS_NEW_API
-        bool "Enables Litepath New API in Wave500"
-        depends on PPA_PUMA7
-        default n
-        ---help---
-           Configures Litepath feature New API for Wave500. This
-           symbol enabled the WiFi Access Point driver to make use
-           of the Directpath extended API exposed by LitePath for
-           Partial Offload.
 endmenu
 
 menu  "Intel PUMA7 AVALANCHE support"
--- a/drivers/net/Kconfig
+++ b/drivers/net/Kconfig
@@ -389,14 +389,25 @@ config VMXNET3
 	  To compile this driver as a module, choose M here: the
 	  module will be called vmxnet3.
 
-config LPAL_DIRECTPATH
-        tristate "LitePath DirectPath Interface"
-        depends on PPA_PUMA7
-        default n
-        help
-          Partial Offload SUpport for WiFi. This turns on the API
-          support for LitePath API library to be used by WiFi
-          Access Point driver.
+config LTQ_PPA_API_DIRECTPATH
+	bool "Enables Litepath API"
+	depends on PPA_PUMA7
+	default n
+	help
+	  Configures API support for partial offload. This symbol
+	  enables the WiFi Access Point driver to make use of
+	  the Directpath API exposed by LitePath for Partial
+	  Offload.
+
+config LTQ_PPA_API_DIRECTPATH_HAS_NEW_API
+	bool "Enables Litepath Extended API"
+	depends on PPA_PUMA7
+	default n
+	help
+	  Configures new API support for partial offload. This
+	  symbol enables the WiFi Access Point driver to make use
+	  of the Directpath extended API that make use of PUMA7
+	  Partial Offload feature.
 
 source "drivers/net/hyperv/Kconfig"
 
--- a/drivers/net/lpal_api/Makefile
+++ b/drivers/net/lpal_api/Makefile
@@ -1,35 +1,7 @@
-################################################################################
 #
-# Linux driver for VMware's vmxnet3 ethernet NIC.
-#
-# Copyright (C) 2007-2009, VMware, Inc. All Rights Reserved.
-#
-# This program is free software; you can redistribute it and/or modify it
-# under the terms of the GNU General Public License as published by the
-# Free Software Foundation; version 2 of the License and no later version.
-#
-# This program is distributed in the hope that it will be useful, but
-# WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE, GOOD TITLE or
-# NON INFRINGEMENT.  See the GNU General Public License for more
-# details.
-#
-# You should have received a copy of the GNU General Public License
-# along with this program; if not, write to the Free Software
-# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
-#
-# The full GNU General Public License is included in this distribution in
-# the file called "COPYING".
-#
-# Maintained by: Shreyas Bhatewara <pv-drivers@vmware.com>
-#
-#
-################################################################################
-
-#
-# Makefile for the LitePath Directpath driver
+# Makefile for the LitePath Adaptation Layer driver
 #
 
-obj-$(CONFIG_LPAL_DIRECTPATH) += lpal_dp.o
+obj-$(CONFIG_LTQ_PPA_API_DIRECTPATH_HAS_NEW_API) += lpal_dp.o
 
 lpal_dp-objs := lpal_api.o lpal_directpath.o
